### PR TITLE
Add file type XDP

### LINKED ISSUE #636: File extensions
Up to now, we listed all the file types/extension https://github.com/SAP/abap-file-formats/blob/main/docs/specification.md#File-Extensions

Somehow we missed XFA-3_3, see SFPF https://github.com/SAP/abap-file-formats/tree/main/file-formats/sfpf

### PR DESCRIPTION
**Pull request by albertmink**
fixes #636

### CODE REVIEW

File: docs/specification.md [hunk 1]
```diff
@@ -54,12 +54,13 @@ For the text elements of classes, this language will be the original language (p
...
 ## File Extensions
 
-ABAP file formats define the file types:
+ABAP file formats use the file types:
 * **`.abap`** stores ABAP source code as plain text
 * **`.acds`** stores source code of ABAP Core Data Services (CDS) as plain text
 * **`.json`** stores content of form-based editors or properties of objects, such as `originalLanguage`, `abapLanguageVersion` and others.
 Details are found in [JSON Files in ABAP File Formats](./json.md)
 * **`.properties`** stores text elements as plain text and [translation texts](properties.md)
+* **`.xdp`** stands for XML Data Packaging and is part of Form Objects (SFPF)
```
Comment (wurzka) [inline docs/specification.md:63]: There is one more missing:
- https://github.com/SAP/abap-file-formats/tree/main/file-formats/csnm

**Review by wurzka — COMMENTED** [review]
Is it the idea to list all extensions here? Then, there is one more missing, see my comment

File: docs/specification.md (hunk 1 above)
Comment (albertmink) [reply docs/specification.md:63]: I believe the list of file types contributes to a certain transparency of the project.

File: docs/specification.md [hunk 2]
```diff
@@ -54,12 +54,14 @@ For the text elements of classes, this language will be the original language (p
...
+* **`.csn`** is a notation for compact representations of CDS models
```
Comment (wurzka) [inline docs/specification.md:64]: Should we also add that it's part of CSN Models (CSNM) like above?
```suggestion
* **`.csn`** is a notation for compact representations of CDS models and is part of CSN Models (CSNM)
```

**Review by wurzka — CHANGES_REQUESTED** [review]
Only one more remark

**Review by wurzka — APPROVED** [review]
Looks good to me ☺️

**Pull request merged**